realm/src/main/java/io/realm/Realm.java

```diff
@@ -1756,9 +1751,30 @@ public static synchronized boolean compactRealmFile(Context context, String file
      *
      * @param context an Android {@link android.content.Context}
      * @return true if successful, false if any file operation failed
+     *
+     * @throws IllegalStateException if trying to compact a Realm that is already open
      */
     public static boolean compactRealmFile(Context context) {
-        return compactRealmFile(context, DEFAULT_REALM_NAME);
+        return compactRealmFile(context, DEFAULT_REALM_NAME, null);
+    }
+
+    /**
+     * Compact a realm file. A realm file usually contain free/unused space.
```

Comment:
It's impossible to argue against intuition but the documentation for `deleteRealmFile()` uses capital R ;-)